src/error.rs

```diff
@@ -0,0 +1,73 @@
+// Copyright 2018, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+
+use Error;
+use IOErrorEnum;
+use std::io;
+
+
+impl From<IOErrorEnum> for io::ErrorKind {
+    fn from(kind: IOErrorEnum) -> Self {
+        match kind {
+            IOErrorEnum::Failed => io::ErrorKind::Other,
```

Comment:
Since you return `io::ErrorKind::Other` a lot of times, why not grouping all error returning it?